lib/reverse_proxy_plug.ex

```diff
@@ -412,12 +415,4 @@ defmodule ReverseProxyPlug do
   defp host_header_from_url(%URI{host: host, port: port, scheme: "https"}) do
     "#{host}:#{port}"
   end
-
-  defp http_client do
-    Application.get_env(
-      :reverse_proxy_plug,
-      :http_client,
-      HTTPClient.Adapters.HTTPoison
-    )
-  end
```

Comment:
What's the reason for removing the global configuration? I think this would be quite useful if specifying a client is mandatory.

Reply:
I removed it to enforce extensibility, allowing different adapters for different endpoints/routes.
However, in the way the options are coded now, this global configuration can already be overriden by the options passed to the plug.

Should I allow for a global config, but document better this behavior for people who want to override it?

Also, I think we could enforce it in such a way only raises when both are empty.
WDYT?

Reply:
Yes, both those suggestions sound like a good way to go about it.

Reply:
Ok, so I'll rework this to keep the global config for retrocompatibility, but still allow passing as an "instance option" for the plug. And raise if both are unset

Reply:
@Fedjmike I've brought both the default and the global config back, there's also a test for this

Reply:
@Fedjmike I've brought both the global config and the default back. There's also a test for this!